tasty-query/shared/src/main/scala/tastyquery/Subtyping.scala

```diff
@@ -312,6 +337,9 @@ private[tastyquery] object Subtyping:
           tp2.typeParams.lengthCompare(tp1.paramRefs) == 0
             && isSubtype(tp1.resultType, tp2.appliedTo(tp1.paramRefs))
 
+    case tp1: RefinedType =>
+      isSubtype(tp1.parent, tp2)
```

Comment:
can tp1 have incompatible refinements with tp2? e.g. if they have the same class type, but tp1 introduces a override

Reply:
revisiting, I guess no because refinements have to be subtype of their base class

Reply:
It doesn't matter. A refinement can only *improve* its parent. In other words, it is always true that `P { refinement } <: P`. Therefore, if `P <: tp2`, by transitivity of subtyping it is always correct to say that `P { refinement } <: tp2`.

(if it does have incompatible with *itself*, it effectively becomes `Nothing`, and so it is also trivially a subtype of `tp2`, so this is fine as well)